src/sources.py

```diff
@@ -284,19 +284,54 @@ def parse_obj_arr_obj(obj, c_file, prefix, obj_typename):
         condition = "\n                && ".join( \
             ['strcmp (tree->u.object.keys[i], "%s")' % i.origname for i in obj.children])
         c_file.write("""
-    if (tree->type == yajl_t_object && (ctx->options & OPT_PARSE_STRICT))
-      {
+    if (tree->type == yajl_t_object) {
         size_t i;
-        for (i = 0; i < tree->u.object.len; i++)
-          {
-            if (%s && ctx->errfile != NULL)
-              {
-                (void) fprintf (ctx->errfile, "WARNING: unknown key found: %%s\\n",
-                        tree->u.object.keys[i]);
-              }
-          }
-      }
-""" % condition)
+        size_t j = 0;
+        size_t cnt = tree->u.object.len;
+        yajl_val resi = NULL;
+
+        if (ctx->options & OPT_PARSE_FULLKEY) {
+            resi = calloc(1, sizeof(*tree));
+            if (resi == NULL) {
+                free_%s(ret);
+                return NULL;
+            }
+            resi->type = yajl_t_object;
+            resi->u.object.keys = calloc(cnt, sizeof(const char *));
+            if (resi == NULL) {
+                free_%s(ret);
+                yajl_tree_free(resi);
+                return NULL;
+            }
+            resi->u.object.values = calloc(cnt, sizeof(yajl_val));
+            if (resi == NULL) {
```

Comment:
same here, should it check for `resi->u.object.values == NULL`?

Reply:
fixed